app/src/components/search/filters/selectFilterModal.tsx

```diff
@@ -0,0 +1,232 @@
+    {
+      filter,
+      onOpen: parentOnOpen,
+      onClose: parentOnClose,
+      selected,
+      setSelected,
+    },
+    headingRef?
+  ) => {
+    const {
+      isOpen,
+      onOpen: modalOnOpen,
+      onClose: modalOnClose,
+    } = useDisclosure();
+
+    const [searchText, setSearchText] = useState("");
+    const [currentPage, setCurrentPage] = useState(1);
+    const itemsPerPage = 10;
+
+    useEffect(() => {
+      setCurrentPage(1);
+    }, [searchText]);
+
+    const filteredOptions = filter.options.filter((option) =>
+      option.name.toLowerCase().includes(searchText.toLowerCase())
+    );
+
+    const pageCount = Math.ceil(filteredOptions.length / itemsPerPage);
+
+    const currentOptions = filteredOptions.slice(
+      (currentPage - 1) * itemsPerPage,
+      currentPage * itemsPerPage
+    );
+
+    const handleOpen = () => {
+      modalOnOpen();
+      parentOnOpen();
+    };
+
+    const handleClose = () => {
+      modalOnClose();
+      parentOnClose();
+    };
+
+    return (
+      <>
+        <Button
+          buttonType="link"
+          id="modal-btn"
+          onClick={handleOpen}
+          sx={{
+            textDecoration: "none",
+            padding: 0,
+          }}
+        >{`View all ${filter.name.toLowerCase()}${
+          filter.name === "Publishers" ? "" : "s"
+        }`}</Button>
+        <Modal
+          finalFocusRef={headingRef as React.RefObject<FocusableElement>}
+          isOpen={isOpen}
+          onClose={handleClose}
+        >
+          <ModalOverlay />
+          <ModalContent sx={{ borderRadius: "4px" }}>
+            <Box
+              sx={{
+                bg: "ui.bg.default",
+                borderRadius: "4px 4px 0px 0px",
+                borderBottom: "1px solid",
+                borderColor: "ui.border.default",
+                marginBottom: "s",
+              }}
+            >
+              <Heading
+                size="heading5"
+                paddingTop="xs"
+                paddingLeft="s"
+                marginBottom="xs"
+              >
+                {`${filter.name}${filter.name === "Publishers" ? "" : "s"}`}
+              </Heading>
+            </Box>
+            <ModalBody>
+              <DCSearchBar
+                id={`search-${filter.name}-options`}
+                labelText={`Search ${filter.name}${
+                  filter.name === "Publishers" ? "" : "s"
+                }`}
+                maxWrapperWidth="400px"
+                textInputProps={{
+                  id: "filter-search-text",
+                  isClearable: true,
+                  labelText: `Search ${filter.name}${
+                    filter.name === "Publishers" ? "" : "s"
+                  }`,
+                  name: "filter_keywords",
+                  placeholder: `Search ${filter.name.toLowerCase()}${
+                    filter.name === "Publishers" ? "" : "s"
+                  }`,
+                  onChange: (e) => setSearchText(e.target.value),
```

Comment:
I think this needs a `clear` callback? If you try to search something and presses the "X" button, it clears the field but doesn't reset the modal.